classes/class-wc-connect-nux.php

```diff
@@ -98,6 +98,37 @@ public function register_add_service_to_zone_pointer( $pointers ) {
 			return $pointers;
 		}
 
+		public static function get_banner_type_to_display( $status = array() ) {
+			/* The NUX Flow:
+			- Case 1: Jetpack not connected (with TOS or no TOS accepted):
+				1. show_banner_before_connection()
+				2. connect to JP
+				3. show_banner_after_connection(), which sets the TOS acceptance in options
+			- Case 2: Jetpack connected, no TOS
+				1. show_tos_only_banner(), which accepts TOS on button click
+			- Case 3: Jetpack connected, and TOS accepted
+				This is an existing user. Do nothing.
+			*/
+			switch ( $status['jetpack_connection_status'] ) {
+				case self::JETPACK_UNINSTALLED:
+				case self::JETPACK_INSTALLED:
+				case self::JETPACK_ACTIVATED:
+					return 'before_jetpack_connection';
+				case self::JETPACK_CONNECTED:
+					// Has the user just gone through our NUX connection flow?
+					if ( $status['should_display_after_cxn_banner'] ) {
+						return 'after_jetpack_connection';
+					}
+					// Has the user already accepted our TOS? Then do nothing.
+					// Note: TOS is accepted during the after_connection banner
+					if ( ! $status['tos_accepted'] ) {
+						return 'tos_only_banner';
+					}
+				default:
+					return false;
+			}
+		}
+
 		public function get_jetpack_install_status() {
 			// check if Jetpack is activated
 			if ( ! class_exists( 'Jetpack_Data' ) ) {
```

Comment:
If you first check `validate_plugin()`, you can make this function a little flatter and avoid else-statements:
```
if( 0 !== validate_plugin() ) return JETAPCK_UNINSTALLED;
if ( ! class_exists() ) return JETPACK_INSTALLED;
if( defined() ) return dev
```